crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -953,12 +953,7 @@ impl IdentityManager {
         }
 
         if !users_with_no_devices_on_unfailed_servers.is_empty() {
-            info!(
-                ?users_with_no_devices_on_unfailed_servers,
-                "Waiting for `/keys/query` to complete for users who have no devices"
-            );
```

Comment:
We should only log something here if we actually end up waiting for one or more `/keys/query` operations, which we will only do if one of the users in `users_with_no_devices_on_unfailed_servers` has been flagged as having an outdated devicelist.

`self.get_updated_keys_for_user` will return `Some(..)` for such users, so instead of logging upfront like this, we collect a list below and log after the wait.